src/execution/__tests__/abstract-test.ts

```diff
@@ -271,7 +271,7 @@ describe('Execute: Handles execution of abstract types', () => {
       errors: [
         {
           message:
-            'Abstract type "Pet" must resolve to an Object type at runtime for field "Query.pet". Either the "Pet" type should provide a "resolveType" function or each possible type should provide an "isTypeOf" function.',
+            'Abstract type resolution for "Pet" for field "Query.pet" failed. Encountered abstract type "Pet" must resolve to an Object or Interface type at runtime. Either the "Pet" type should provide a "resolveType" function or each possible type should provide an "isTypeOf" function.',
```

Comment:
The error messages have to be changed (in my opinion) because there is now a multi-step process for deriving the runtime type, and we want to be as helpful as possible by pointing out what type we started with and where we failed. It's not a separate change => it's just part and parcel of what we're doing here, it wouldn't make sense on its own, it would just be repetitive unless we have the multi-step process.

Reply:
> and we want to be as helpful as possible by pointing out what type we started with

Please note that this is not a **client** error, the client can't trigger that error unless there is a bug in type resolvers. So for a client having this info is useless and it should matter only for people working on a schema.

Moreover, for clients, this detail is already accessible through the `path` and `locations`.
GraphiQL will highlight the field you get an error on and you will see its type in a tooltip.

I'm not sure having it explicitly in the message just for schema developers justifies an additional complexity of passing additional arguments just for error reporting.

In general, type resolvers should be independent of each other same as field resolvers. So it shouldn't matter what was previous type resolver since it's doesn't affect the resolver's argument.
If the schema author really wants to debug it when he can easily use the available type name and field name to quickly find initial type in a chain.

Reply:
From my survey of the repo and at GraphQL-Tools, there are many schema developers out there who get confused by cryptic error messages from this (and many other!) packages. I am in favor of keeping more details in the error message, perhaps I can refactor per @saihaj  suggestion to just pass around one object that bundles some of the constant parameters?

Reply:
Actually, I think it just makes sense to go with the shorter messages. We can always change it later.